primitives/core/src/traits.rs

```diff
@@ -187,20 +192,143 @@ pub trait BareCryptoStore: Send + Sync {
 	/// the public key and key type provided do not match a private
 	/// key in the keystore. Or, in the context of remote signing
 	/// an error could be a network one.
-	fn sr25519_vrf_sign(
+	async fn sr25519_vrf_sign(
 		&self,
 		key_type: KeyTypeId,
 		public: &sr25519::Public,
 		transcript_data: VRFTranscriptData,
 	) -> Result<VRFSignature, Error>;
 }
 
-/// A pointer to the key store.
-pub type BareCryptoStorePtr = Arc<parking_lot::RwLock<dyn BareCryptoStore>>;
+/// A wrapper around CryptoStore to be used in synchronous fashion.
+pub struct SyncCryptoStore(CryptoStorePtr);
```

Comment:
I still imagine something like this: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5a38cee61c1b26af7d1928d636cd8ab6

Reply:
ahh TIL. But isn't it a bit more verbose to implement the methods twice:
```
impl SyncTest for dyn Test {
    fn sync_test(&self) {
        futures::executor::block_on(self.test())
    }
}

impl<T: Test> SyncTest for T {
    fn sync_test(&self) {
        futures::executor::block_on(self.test())
    }
}
```
?

Reply:
I am in favor of both `CryptoStore` as well as `SyncCryptoStore` being a trait, especially with the latter being an auto trait of the former. I am not sure whether each `SyncCryptoStore` has to be prefixed by `sync_`, but I guess that is an aesthetic question.

Reply:
Yeah we probably don't need the prefixing with `sync_`. I first had the `Test` trait being a super trait of the `SyncTest`, but yeah that is not required now anymore ;) 

> ahh TIL. But isn't it a bit more verbose to implement the methods twice:

I'm still in favor. Anyway, we probably only require the implementation for `dyn Test`.

Reply:
I was curious about whether it's possible to have 2 methods with the same name, one async and the other is sync. 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0487bcd2f2ce0389b0fef5acbd404b8f

Doesn't seem to be possible so i guess the `sync_` prefix is required here.

Reply:
@rakanalh any updates on making `SyncCryptoStore` a trait?

Reply:
> I was curious about whether it's possible to have 2 methods with the same name, one async and the other is sync.
> https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0487bcd2f2ce0389b0fef5acbd404b8f
> 
> Doesn't seem to be possible so i guess the `sync_` prefix is required here.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e95e6c26ab8c80fe349df1e2d41cddd3

Reply:
Done